### PR TITLE
creating a copy of the notebooks, and changing the tariff used from CNFL to ICE

### PR DESCRIPTION
**Pull request by arnavgautam**
Hi Jose,

Wanted to use this to give some updates, and ask some questions.
I've spent some time going through the codebase, to try and understand the individual modules as well as how the data and functions all tie together. I've also set up my local environment to be able to run all of this, which has helped.

I then started trying to get a new set of notebooks running for ICE as opposed to CNFL, but it's been a bit more complicated than I expected. I was able to determine where we pass in the tariff data used, and have a parallel flow set up that uses the ICE values created in tariff_data.jl. However, I haven't yet been able to determine where in the flow the usage data itself enters. At first I was exploring the hist_CNFL_res.csv and such files, but those don't seem to be used anywhere outside of your sandbox.

I'll keep digging into this, but if you have any tips on how I can tease the CNFL codepath out of the more neutral code, it would be appreciated.

Thanks!
-Arnav

P.S. I spent some time developing on a clone of the repo, but didn't realize I wasn't a contributor so I couldn't actually push branches. So I've made this fork just now which I don't mind working off of, but it might be easier to coordinate if I were working off the repo itself. I can make side branches in order to not break functionality.

### CODE REVIEW

**Comment by jd-lara** [comment]
Hi Arnav, it is better to update one notebook at the time in order to not lose track of the whole flow. Let me clarify some thing. 

You don't need to replicate the function PVCR, we don't have to create a parallel set-up for the ICE tariffs. In the data folder there is already data for ICE costumers, the functions take Tariff types and perform the operations. What you need to make are new consumers in `consumer_data.jl` that take XXX_ICE tarrifs. 

If you see in the file `Module1_DG_tariff_application.ipynb` the function `annual_energy_balance` takes the residential from `consumer_data.jl` so the change you need to do is: rename the variable `residential` to `residential_CNFL` and make a new variable `residential_ICE`. In this way you can run 

```Julia
base_ebalance = annual_energy_balance(residential_CNFL, PVRes, print_output=false);
```

In one the current norebook `Module1_DG_tariff_application.ipynb`that should be renamed ``Module1_DG_tariff_application_CNL.ipynb` and then 

```Julia
base_ebalance = annual_energy_balance(residential_ICE, PVRes, print_output=false);
```

In a new notebook copy of `Module1_DG_tariff_application.ipynb` and renamed `Module1_DG_tariff_application_ICE.ipynb`

The same principle applied to the other modules, they are already general and what we needt is to create new data items for ICE when needed and rename those that are not specific yet. All the code base works for the types created in the `types` folder and we use composition to collect them. 

We can talk it over the phone sometime this week if it isn't clear.

File: data/ICE_consumer_data.jl [hunk 1]
```diff
@@ -0,0 +1,28 @@
+residential = Residential(20, 550, -0.005, nothing,
```
Comment (jd-lara) [inline data/ICE_consumer_data.jl:1]: All of the content of this file can be copied into `https://github.com/jd-lara/PVCR/blob/master/data/consumer_data.jl` and then just rename the variables.

File: ICE_PVCR.jl [hunk 2]
```diff
@@ -0,0 +1,66 @@
+using DelimitedFiles
```
Comment (jd-lara) [inline ICE_PVCR.jl:1]: PVCR is the equivalent of a `main` file it should only be one for the whole repo. The only files that need to be duplicated in the whole repo are the notebooks.

**Review by jd-lara — CHANGES_REQUESTED** [review]

**Comment by arnavgautam** [comment]
Closing this pull request as it's been made obsolete by the per-notebook requests

**Pull request closed without merging**